.github/workflows/bench.yml

```diff
@@ -0,0 +1,63 @@
+# Based on https://github.com/actions-rs/meta/blob/master/recipes/msrv.md
+
+on: [ pull_request ]
+
+name: Benchmark
+
+jobs:
+  benchmark:
+    name: Run benchmarks
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - stable
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Set env vars
+        run: |
+          THIS_SHA=$(echo $GITHUB_SHA | cut -c 1-7)
+          MAIN_SHA=$(echo $(git rev-parse main) | cut -c 1-7)
+
+          echo "THIS_BENCH=${THIS_SHA}.bench" >> $GITHUB_ENV
+          echo "MAIN_BENCH=${MAIN_SHA}.bench" >> $GITHUB_ENV
+
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          override: true
+        
+      - name: Install critcmp
+        run: cargo install critcmp
```

Comment:
I would add the `rust-cache` action in, it should cache cargo binaries as well as dependencies to really speed things along.